pydantic/networks.py

```diff
@@ -96,7 +96,31 @@ def __hash__(self) -> int:
     @property
     def defined_constraints(self) -> dict[str, Any]:
         """Fetch a key / value mapping of constraints to values that are not None. Used for core schema updates."""
-        return {field.name: getattr(self, field.name) for field in fields(self)}
+        return {field.name: value for field in fields(self) if (value := getattr(self, field.name)) is not None}
+
+    def __get_pydantic_core_schema__(self, source: Any, handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
+        schema = handler(source)
+
+        if schema['type'] == 'function-wrap':
+            # for function-wrap schemas, url constraints is applied to the inner schema
+            inner = schema['schema']
+
+            if annotated_type := inner['type'] not in ('url', 'multi-host-url'):
+                raise PydanticUserError(
```

Comment:
I think you could simplify here:

```suggestion
        if schema['type'] == 'function-wrap':
            # for function-wrap schemas, url constraints is applied to the inner schema
            inner = schema['schema']

            if annotated_type := inner['type'] in ('url', 'multi-host-url'):
                schema = inner
```

Reply:
Hmm, I don't think this works

Reply:
Because we still want to return `schema`, just with `inner` reassigned after modification. We still need the `function-wrap`